.github/workflows/main.yml

```diff
@@ -16,6 +16,8 @@ jobs:
         with:
           path: ~/.npm
           key: ${{ runner.os }}-${{ hashFiles('**/package-lock.json') }}
+      - name: Upgrade NPM
+        run: npm install -g npm
```

Comment:
Why the pipeline needs to update the NPM each time? Can you add more light to this?

Reply:
Can you update node version instead? node@14 gets installed with npm@6, but node@18 with npm@9 👍 
https://nodejs.org/ro/download/releases

Reply:
This has been fixed. Some extra packages are required to be installed in order to use node v18 (v16 as well)

Reply:
_You have jsdom (jest) dependent on an old version of canvas that tries to build under nodejs@18. Current fix is a workaround, it's better to update dependencies._
originally mentioned by @davidyuk

Reply:
`@vue/cli-plugin-unit-jest` doesn't support jest@28/29.  [see](https://github.com/vuejs/vue-cli/issues/7129)

Reply:
Can you update jsdom only? The `canvas` package is only needed to emulate it for tests

Reply:
Updating `canvas` fixes the issue

Reply:
Maybe `canvas` is not built for the env you have in gh actions 🤷‍♀️

Reply:
Turns out `npm ci` doesn't run locally either. So the issue still remains.
Yes, canvas is trying to build it self because it doesn't a prebuilt version, but then it fails. The packages I had added solved this.
This is also an official fix [[see](https://github.com/Automattic/node-canvas#compiling)]
I think adding the packages is the best option here